h2o-core/src/main/java/hex/grid/HyperSpaceWalker.java

```diff
@@ -333,12 +260,52 @@ protected int integerHash(int[] ar) {
         hashMe[i] = ar[i] * _hyperParams.get(_hyperParamNames[i]).length;
       return Arrays.deepHashCode(hashMe);
     }
+
+    private void validateParams() {
+      // if a parameter is specified in both model parameter and hyper-parameter, this is only allowed if the
+      // parameter value is set to be default.  Otherwise, an exception will be thrown.
```

Comment:
I think that this restriction is unnecessary and overly complicated.
Why don't we just say that hyper-parameters take precedence over model parameters. Then everything is simple, people can specify a param in both if they want, we'll just use the one in the hyperparams... anyway user can access the params from the trained model later, no cheating.
@michalkurka  any thoughts?

Reply:
I think it is good to have this fail-safe mechanism. People make mistakes all the time and experiment with different parameter combinations. It is better to avoid surprises.